pkg/config/config_template.yaml

```diff
@@ -2843,6 +2843,17 @@ api_key:
 #
 # ad_config_poll_interval: 10
 
+## @param ad_allowed_env_vars - list of strings - optional
+## @env DD_AD_ALLOWED_ENV_VARS - list of strings - optional
+## The list of environment variables that are allowed to be resolved in check
+## configurations.
+## The default is to allow all environment variables. If the list is set, but
+## empty, no environment variables will be resolved.
```

Comment:
Having different behavior depending on the value type is usually not a great idea since it's not obvious to newcomers. Also, once loaded in the code the diff in Go between an empty slice and a nil slice is error prone.

I think a dedicated `ad_disable_env_vars` would be more explicit and less error prone. It would also allow us to add a default to both in the code. Not having default means the config can't know what to expect for a setting which makes validation impossible.

Reply:
I think this should be an allow-list instead of a block-list. The goal is to let users control what envs can be used. It’s easier to list what you want to allow than to block everything you don’t need.

The main question is what to do with an empty allow-list. I think we can set a default like `[]string{}`, as you mentioned in the other comment. We should only use this setting if the user has set it (we can check with `IsConfigured`). If it’s not configured, we keep the current behavior. If the list is set but empty, it means no environment variables should be resolved. Since this is an allow-list, an empty list should block everything if set explicitly.

Reply:
I agree that an allow-list is better. I'm saying that an empty list to disable env vars resolution is not a intuitive behavior for customers and is error prone. We had such behavior in the past and it's very easy when working with YAML to comment a few line and create en empty list. We also had different YAML lis produce different results when loading an empty YAML list.

I would ignore a nil or empty list for `ad_allowed_env_vars` and add another dedicated setting to disable env var resolution.